code/game/gamemodes/wizard/spellbook.dm

```diff
@@ -95,6 +96,10 @@
 		available_artifacts.Remove(T) //Remove the path from the list
 	//Result is a list full of /datum/spellbook_artifact objects
 
+	if(Holiday == "Christmas")
```

Comment:
You could do this a bit more OOP if you set a `Holiday_required` for each spell and checked for it when giving out spells.

That way, you can also add Halloween spells without too much hassle, for example.